pyccel/codegen/printing/cwrappercode.py

```diff
@@ -87,7 +87,11 @@ def function_signature(self, expr):
     def get_declare_type(self, expr):
         dtype = self._print(expr.dtype)
         prec  = expr.precision
-        dtype = self.find_in_dtype_registry(dtype, prec)
+        if expr.rank > 0 and self._target_language == 'c' and not expr.is_pointer:
+            self._additional_imports.add("ndarrays")
```

Comment:
Maybe it would be better to do:
```
if self._target_language == 'c':
    return CCodePrinter.get_declare_type(self, expr)
```
So we don't need to maintain this condition in 2 places

Reply:
we can't handle all c stuff in the `CCodePrinter`. that why we create `cwrappercode ` for things related to wrapper.

Reply:
Then can you explain why ndarrays only needs including in the wrapper?

Reply:
we are using ndarray type, so we need to include the header.

Reply:
But this is also the case outside the CWrapper isn't it? The aim is for the `CWrapperCodePrinter` to fall back to the `CCodePrinter` wherever possible as this minimises code duplication and ensures that if something changes in the condition (e.g. we decide to have an ndarray file per rank, or to split ndarray somehow) then we only need to fix it in one place. So `CWrapperCodePrinter` should only implement more complicated logic if the equivalent does not exist in the `CCodePrinter`.

This function is already larger than I would like for something which has a direct equivalent in the `CCodePrinter`. I created the function because I needed to avoid the different handling of the rank, but you can already see the danger of it not falling back to c from the fact that the cwrapper is potentially missing the `stdint` import.

If the import of `ndarrays` here is something that only needs doing in the cwrapper then the code is fine as it is. If it is specific to c then calling `CCodePrinter.get_declare_type` is safer

Reply:
I tried to remove these conditions but I faced many problems because we don't have a proper type of t_ndarray in the ast to differentiate between pyccel array object and ndarray type.

Reply:
Presumably the `is_ndarray` property is not sufficient for this?

Reply:
in ccode it's sufficient, but not in the wrapper, because `pyccelarrayobject` has the same properties of ndarray.
I see that we need to make the NativeGeneric type in such a way that can accept non-standard types (ndarray).
what do you think?